DA-15: Improve timeline view and add visualization config in scheme

JeannedArk opened the pull request:
The width calculation in the timeline is now dynamically calculated depending
on the max width of the neighbour nodes. Subgraphs are displayed in their own
section. Additionally the visualization elements can be configured in the scheme.
The kind if visualization (timeline and graph) can be configured also in the
scheme builder.

JeannedArk committed:
DA-15: Improve timeline view and add visualization config in scheme

The width calculation in the timeline is now dynamically calculated depending
on the max width of the neighbour nodes. Subgraphs are displayed in their own
section. Additionally the visualization elements can be configured in the scheme.
The kind if visualization (timeline and graph) can be configured also in the
scheme builder.

JeannedArk commented:
Please test and do not use the existing schemes.

NicklasLinz commented:
I think this should not be on the master until it is fully implemented (ordering of link labels, etc.)

JeannedArk committed:
DA-15: Allow visualization configuration in scheme of none, graph and timeline

The scheme allows to define different visualizations of none, graph and timeline.
Furthermore display options were added for e.g. positioning the links in the
timeline mode. Therefore the scheme builder and the scheme preview was adjusted.
The entity models were changed so that every label and linklabel are now identified
by there id and not by there name/ string label, so that configurations to each
label can be applied.
The timeline places the nodes depending on the given configuration.
The visualization options are no exclusive.
Some bugs were fixed according to the displaying of the visualization kinds.
schemeUploadModal html and controller was refactored.
